src/OrleansServiceBus/Providers/Streams/EventHub/EventHubQueueCache.cs

```diff
@@ -148,7 +159,62 @@ public bool TryGetNextMessage(object cursorObj, out IBatchContainer message)
 
     }
 
-    internal class AveragingCachePressureMonitor
+    public class SlowConsumingPressureMonitor : ICachePressureMonitor
```

Comment:
Suggest AveragingCachePressureMonitor be broke into it's own file, and added in the default cache creation logic, like other ICachePressureMonitors.